pkg/deployments/scaler.go

```diff
@@ -33,12 +33,15 @@ func (s *scaler) AtLeastOne() {
 	}
 }
 
-// UpdateState updates the current state of the scaler and
-// scales if needed.
+// UpdateState updates the current state of the scaler
 func (s *scaler) UpdateState(replicas, min, max int32) {
 	log.Printf("UpdateState(%v, %v, %v)", replicas, min, max)
-	s.setMinMax(min, max)
-	s.compareScales(replicas, -1)
+	s.mtx.Lock()
+	defer s.mtx.Unlock()
+	s.minScale = min
+	s.maxScale = max
+	s.currentScale = replicas
+	s.desiredScale = -1
```

Comment:
This line clears the desired scale and is called by reconcile. This resets the state on all instances and fixes the stale state problem for ex-leader instances. Nevertheless this is not ideal for the current leader as the desired state may be set already.